### PR TITLE
Problem (Fix #1621): Can't deposit into other's new staking address

### LINKED ISSUE #1621: Problem: Can't deposit into other's new staking address
The intension of the restriction is to prevent wrong deposit tx, but it also block valid cases.

Options:
- remove the restriction
- add confirmation in client ui
- whitelist the operator

Decision: Option2
- Remove the restriction from client network ops
- add a check and double confirmation in client-cli.

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- Remove the restriction from client network ops code
- Add check and double confirmation in client-cli

### CODE REVIEW

File: client-network/src/network_ops/default_network_ops_client.rs [hunk 1]
```diff
@@ -137,20 +137,7 @@ where
...
-        // if the to_address belongs to current wallet, we do not check the state
-        let staking_addresses = self.wallet_client.staking_addresses(name, enckey)?;
-        if !staking_addresses.contains(&to_address) {
-            let staked_state = self.get_staked_state(name, &to_address, verify_staking)?;
-            verify_unjailed(&staked_state).map_err(|e| {
-                Error::new(
-                    ErrorKind::ValidationError,
-                    format!("Failed to validate staking account: {}", e),
-                )
-            })?;
```
Comment (tomtau) [inline client-network/src/network_ops/default_network_ops_client.rs:151]: maybe the jailing test could be optionally run (if the state exists?)

Comment (yihuang) [reply client-network/src/network_ops/default_network_ops_client.rs]: Done

**Comment by yihuang** [comment]
> maybe also bump version numbers in the affected client-* Cargo.toml

Done

**Comment by tomtau** [comment]
> > maybe also bump version numbers in the affected client-* Cargo.toml
> 
> Done

client-rpc is also affected if client-network changed?

**Comment by yihuang** [comment]
> > > maybe also bump version numbers in the affected client-* Cargo.toml
> > 
> > 
> > Done
> 
> client-rpc is also affected if client-network changed?

Yes, just bumped the version too, and reflected that in CHANGELOG.

**Pull request merged**